src/status_im/contexts/profile/edit/modal/view.cljs

```diff
@@ -0,0 +1,266 @@
+                                    name-too-short? :default
+                                    :else           :success)
+            info-message          (or validation-msg
+                                      (i18n/label :t/minimum-characters
+                                                  {:min-chars profile-validator/min-length}))
+            full-name             (if (seq display-name)
+                                    display-name
+                                    initial-display-name)]
+        [rn/scroll-view
+         {:on-layout               set-scroll-height
+          :on-scroll               set-scroll
+          :scroll-event-throttle   64
+          :content-container-style {:flex-grow 1}}
+         [rn/view {:on-layout set-content-container-height}
+          [rn/view {:style style/content-container}
+           [quo/text
+            {:style  style/title
+             :size   :heading-1
+             :weight :semi-bold}
+            (i18n/label :t/edit-profile)]
+           [rn/view {:style style/input-container}
+            [rn/view {:style style/profile-input-container}
+             [quo/profile-input
+              {:customization-color custom-color
+               :placeholder         initial-display-name
+               :on-press            open-picture-picker
+               :image-picker-props  {:profile-picture     profile-image
+                                     :full-name           full-name
+                                     :customization-color custom-color}
+               :title-input-props   {:default-value  ""
+                                     :auto-focus     true
+                                     :max-length     c/profile-name-max-length
+                                     :on-change-text on-change-text}}]]
+            [quo/info-message
+             {:status          info-type
+              :size            :default
+              :icon            (if valid-name? :i/positive-state :i/info)
+              :color           (when (= :default info-type) colors/white-70-blur)
+              :container-style style/info-message}
+             info-message]
+            [quo/text
+             {:size   :paragraph-2
+              :weight :medium
+              :style  style/color-title}
+             (i18n/label :t/colour)]]]
+          [quo/color-picker
+           {:blur?            true
+            :default-selected custom-color
+            :on-change        set-custom-color
+            :window-width     window-width
+            :container-style  {:padding-left (responsiveness/iphone-11-Pro-20-pixel-from-width
+                                              window-width)}}]]]))))
+
+(defn- submittable-state?
+  [{:keys [new-color? new-picture? new-name? valid-name?]}]
+  (let [acceptable-name? (or (not new-name?) valid-name?)]
+    (or (and new-color? acceptable-name?)
+        (and new-picture? acceptable-name?)
+        (and new-name? valid-name?))))
```

Comment:
If this predicate is not satisfied, the button to continue keeps disabled, meaning no updates to the profile were performed. The user can always just close the modal or navigate back